### PR TITLE
Improved Documentation, Rust Tokenize Shuffle

### PR DESCRIPTION
**Pull request by afang-story**
Added rust tokenize shuffle + commands
Explained exp_data
Fixed commands for training and evaluation
Included examples in the above sections

Combined above examples with some additional commands to have a brief list of commands to download DCLM-baseline, tokenize shuffle, train, and evaluate a 1B_1x model.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -293,6 +392,60 @@ We provide multiple datasets, both as starting points for each of the competitio
...
+--tokenizer "EleutherAI/gpt-neox-20b" \
+--seqlen 2049 \
+--wds-chunk-size 8192 \
+--num-local-cells 512
+
+popd
+```
+
+### Training
+data-config comes from the json created (manually for rust code, automatically for ray) after tokenize shuffle.
```
Comment (GeorgiosSmyrnis) [inline README.md:437]: Is it possible to incorporate the manifest creation in the above?

File: README.md [hunk 2]
```diff
@@ -293,6 +392,60 @@ We provide multiple datasets, both as starting points for each of the competitio
...
+mkdir dclm_tokshuf
+
+cargo run --release -- \
+--input dclm_local \
+--local-cell-dir tokshuf_tmp \ 
+--output dclm_tokshuf \
+--tokenizer "EleutherAI/gpt-neox-20b" \
+--seqlen 2049 \
+--wds-chunk-size 8192 \
+--num-local-cells 512
```
Comment (GeorgiosSmyrnis) [inline README.md:431]: Should `--num-local-cells` be equal to the number of available cores (in which case we should probably decrease the default / add a comment)

File: README.md (hunk 1 above)
Comment (afang-story) [reply README.md:437]: Would have to write a separate python script and have a bash script that runs the two together.

File: README.md (hunk 2 above)
Comment (afang-story) [reply README.md:431]: No. Theres an explanation for this in the rust tokenize shuffle readme, but this default is reasonable.

File: README.md [hunk 3]
```diff
@@ -293,6 +392,60 @@ We provide multiple datasets, both as starting points for each of the competitio
...
+
+popd
+```
+
+### Training
+data-config comes from the json created (manually for rust code, automatically for ray) after tokenize shuffle.
+```bash
+torchrun --nproc-per-node 8 -m training.train --scale 1b_1x_fast --data-config exp_data/datasets/tokenized/dclm_gs3_ls1_rs_tokshuf.json --logs dclm_rs_tokshuf_training_local_logs --torchcompile
```
[the remark this answers is not in the document]
Comment (afang-story) [reply README.md:439]: Added.

File: README.md (hunk 2 above)
Comment (GeorgiosSmyrnis) [reply README.md:432]: Ok sounds good!

**Pull request merged**